domain/composer.json

```diff
@@ -0,0 +1,21 @@
+{
+    "name": "org/domain",
+    "description": "This extension provides new commands for Drush.",
+    "type": "drupal-drush",
+    "authors": [
+        {
+            "name": "Author name",
+            "email": "author@example.com"
+        }
+    ],
+    "require": {
+        "php": ">=5.6.0"
+    },
+    "extra": {
+        "drush": {
+            "services": {
+                "drush.services.yml": "^9"
```

Comment:
Just so I understand -- this means that to support Drush 8 we need to keep the domain.drush.inc file?

Reply:
My understanding is that drush 8 no longer works with Drupal 8, and the reason it might seem to is that drush 8 will actively look for an instance of drush 9 within the drupal 8 installation, and use that.

The actual requirement here is the 'extra' section including 'drush':
https://docs.drush.org/en/master/commands/#specifying-the-services-file

I have copied the relevant content from the domain/composer.json file into this file: domain/domain/composer.json, which hopefully addresses some of these issues.
It seems possible to me that the former file should not declare itself to be a drupal-module?